ARCHITECTURE.md

```diff
@@ -0,0 +1,167 @@
+# `func_adl_xAOD` Architecture
+
+This file provides a brief overview of how the repository is layed out and how the various bits of code in it cooperate to produce C++ code from a `func_adl` query.
+
+After some general information, more details on how things work follow below.
+
+## General Theory
+
+At the most general level:
+
+1. Client starts with a `func_adl` query represented as a python `ast.AST`. The query is rendered as a python ast by the `func_adl` library.
+1. Client creates a `atlas_xaod_executor` object (or `cms_aod_executor`)
+1. Client calls `apply_ast_transformation` method with the `ast`. This does `ast` -> `ast` transformations that simplify and combine `ast` elements.
+1. Client calls `write_cpp_files` method with the resulting `ast` from the previous step. The output directory where the C++ files can be written
+   need to be given to the method as well.
+1. The C++ files produced are ready to run on the input files.
+
+Though only part of the tests, you can see how the template files are run against a file in the `test/atlas/xaod/LocalFile.py` file, specifically the `execute_result_async` method. Given a `func_adl` ast, it calls `atlas_xaod_executor` as above, and then uses a `docker` image to run against some test files.
```

Comment:
The function has been moved to `tests/utils/base.py`

Reply:
Fixed in next commit